File: Binary Search/4. Running/Running.py
import sys, bisect
sys.stdin = open('Running.inp', 'r')
sys.stdout = open('Running.out', 'w')

a = []

n = int(input())
for i in range(n):
    a.append(int(input()))

a.sort()

r = 0
for x in range(n-2):
    for y in range(x+1, n-1):
        left = 2*a[y] - a[x]
        right = 3*a[y] - 2*a[x]
        
        r += bisect.bisect_right(a, right, y+1) - bisect.bisect_left(a, left, y+1)
print(r)

File: Binary Search/4. Running/Running.cpp
#include <bits/stdc++.h>
using namespace std;

long long n, a[1001], r, u, d;

int main () {
    ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0);
    freopen("Running.inp", "r", stdin);
    freopen("Running.out", "w", stdout);
    
    cin >> n;
    for (int i = 0; i < n; i++) cin >> a[i];
    sort(a, a+n);

    for (int x = 0; x < n-2; x++) {
        for (int y = x+1; y < n-1; y++) {
            u = a[y]*2 - a[x];
            d = a[y]*3 - a[x]*2;
            r += upper_bound(a, a+n, d) - lower_bound(a, a+n, u);
        }
    }
    cout << r;
}